addons/pos_razorpay/static/src/app/services/pos_store.js

```diff
@@ -0,0 +1,52 @@
+import { patch } from "@web/core/utils/patch";
+import { PosStore } from "@point_of_sale/app/services/pos_store";
+
+patch(PosStore.prototype, {
+    async pay() {
+        const currentOrder = this.getOrder();
+        const refundedOrder = currentOrder?.lines[0]?.refunded_orderline_id?.order_id;
+        const razorpayPaymentlines = refundedOrder?.payment_ids.filter(
+            (pi) => pi.payment_method_id.use_payment_terminal === "razorpay"
+        );
+        if (refundedOrder && razorpayPaymentlines?.length === 1) {
+            await super.pay();
```

Comment:
```suggestion
        await super.pay();
        if (refundedOrder && razorpayPaymentlines?.length === 1) {
```
If you call super outside the if, you can remove the else under